src/sage/matrix/seymour_decomposition.pyx

```diff
@@ -902,6 +1016,44 @@ cdef class SubmatrixNode(DecompositionNode):
     pass
 
 
+cdef class DecompositionNodeIndexKey:
+
+    cdef frozenset _core
```

Comment:
Nice. Not sure if I understand the terminology "core" though

Reply:
Maybe I will just call it `_key`?

Reply:
makes sense